headless-services/spring-boot-language-server/src/main/java/org/springframework/ide/vscode/boot/java/livehover/v2/SpringProcessConnectorService.java

```diff
@@ -232,7 +236,7 @@ private void scheduleRefresh(ProgressTask progressTask, SpringProcessParams spri
 			
 			try {
 				progressTask.progressEvent(progressMessage);
-				if(endpoint.equals("metrics") && metricName.equals("memory")) {
+				if(METRICS.equals(endpoint) && MEMORY.equals(metricName)) {
```

Comment:
The `metricName` passed from the API call here is ("memory", "gcPauses"). This is to identify the type of metric and not the actual metricName used for the actuator call. (Probably I should have named it `metricType`. But I wanted to keep the [metricQuery](https://github.com/spring-projects/sts4/blob/b2af04b6cd32b9a7b44af5676ffe10da5d5004ca/vscode-extensions/vscode-spring-boot/lib/api.d.ts#L81) more generic to be used for other metric calls.

The [getMemoryMetrics](https://github.com/spring-projects/sts4/blob/b2af04b6cd32b9a7b44af5676ffe10da5d5004ca/headless-services/spring-boot-language-server/src/main/java/org/springframework/ide/vscode/boot/java/livehover/v2/SpringProcessLiveDataExtractorOverJMX.java#L189) function is where the metric names ("jvm.memory.used, jvm.memory.max") are passed.

Also, with the current implementation, the [retrieveLiveMemoryMetricsData](https://github.com/spring-projects/sts4/blob/b2af04b6cd32b9a7b44af5676ffe10da5d5004ca/headless-services/spring-boot-language-server/src/main/java/org/springframework/ide/vscode/boot/java/livehover/v2/SpringProcessLiveDataExtractorOverHttp.java#L139) has the logic to fetch the heap and non-heap memory metrics together.